### PR TITLE
WIP/DOC: start to document high level design and thinking

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -3,13 +3,343 @@
...
+1. direct access to the control system (i.e. reading / writing single PVs)
+2. device level (i.e. ophyd objects, tango devices)
+3. procedure level (i.e. bluesky plans, recursively procedures of procedures...)
+
+At each scale we depend on the lower level.
+
+In line with standard security practice we want to provide users with
+the minimum level of access that will let them achieve what they need
+to.  To that end we are going to focus on providing remote access at
+the level of plans.
```
Comment (danielballan) [inline docs/source/index.rst:129]:
```suggestion
the procedure leve (4), i.e. the level of bluesky plans.
```

File: docs/source/index.rst [hunk 2]
```diff
@@ -3,13 +3,343 @@
...
+client and the server and gives (remote) user more privileges than
+they strictly need.
+
+
+Review Data
+~~~~~~~~~~~
+
+The proposed method for accessing "prompt" data is an externally accessible
+jupyterhub instance.  We can also use the same processing nodes to manage data
+export to a format of the users choice.
```
Comment (danielballan) [inline docs/source/index.rst:155]:
```suggestion
export to a format of the user's choice.
```

File: docs/source/index.rst [hunk 3]
```diff
@@ -3,13 +3,343 @@
...
+1. CLI that mimics the current experience
+2. QT application
+3. Web application
+4. Autonomous agents
+
+
+RunEngine
+~~~~~~~~~
+
+1. killers vs suspenders
```
Comment (danielballan) [inline docs/source/index.rst:344]: What's a "killer"?

Comment (tacaswell) [reply docs/source/index.rst:344]: a suspender that aborts / stops the current run rather pausing.

Comment (danielballan) [reply docs/source/index.rst:344]:
```suggestion
1. Extending suspenders with "killers" that can abort/stop rather than pause
```

**Pull request merged**